lib/cute_pets.rb

```diff
@@ -6,16 +6,42 @@
 module CutePets
   extend self
 
-  def post_pet()
-    pet = nil
-    if ENV.fetch('pet_datasource').downcase == 'petfinder'
+  def fetch_pet()
+    if ENV.fetch("pet_datasource").downcase == "petfinder"
       pet = PetFetcher.get_petfinder_pet
     else
       pet = PetFetcher.get_petharbor_pet
     end
+  end
+
+  def post_pet()
+    pet = nil
+    pet = fetch_pet
+    # Check that we haven't tweeted this pet before.
+    history_file = ENV.fetch("history_file")
+    history_filepath = File.expand_path(history_file)
+    if File.file?(history_filepath)
+      num_tries = ENV.fetch("number_tries").to_i
+      history = File.readlines(history_filepath)
+      num_tries.times do |i|
```

Comment:
Unused block argument - `i`. You can omit the argument if you don't care about it.